packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda"
+import {Logger} from "@aws-lambda-powertools/logger"
+import {injectLambdaContext} from "@aws-lambda-powertools/logger/middleware"
+import {DynamoDBClient, PutItemCommand} from "@aws-sdk/client-dynamodb"
+import {marshall} from "@aws-sdk/util-dynamodb"
+import middy from "@middy/core"
+import inputOutputLogger from "@middy/input-output-logger"
+import errorHandler from "@nhs/fhir-middy-error-handler"
+import {v4 as uuidv4} from "uuid"
+
+const logger = new Logger({serviceName: "updatePrescriptionStatus"})
+const client = new DynamoDBClient({region: "eu-west-2"})
+const tableName = process.env.TABLE_NAME
+
+const MISSING_FIELDS_RESPONSE: APIGatewayProxyResult = {
+  statusCode: 400,
+  body: JSON.stringify({error: "Missing required fields"}),
+  headers: {
+    "Content-Type": "application/fhir+json",
+    "Cache-Control": "no-cache"
+  }
+}
+
+const lambdaHandler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    // Parse request body
+    const requestBody = JSON.parse(event.body || "")
+
```

Comment:
Just try catch around the JSON parse, then if it errors you can log a nice descriptive error that the body is invalid json and return an appropriate 400 Bad request response back to the client.

Reply:
This try catch looks unchanged, and still too far reaching